Adding the FindSamplesToCompare wdl

meganshand opened the pull request:
This fingerprints an input vcf against a list of known truth samples, then calls Benchmarking wdl with the correct truth data.

kachulis commented on Fingerprints/FindSamplesToCompare.wdl:
```diff
@@ -0,0 +1,581 @@
+
+            Pair[File,File] vcf_and_index_to_compare = select_first([vcf_and_index_symbolic_removed,vcf_and_index_original])
+
+            #call CompareToGroundTruth {
```
can this be removed, or do you want to leave it in?

kachulis commented on Fingerprints/FindSamplesToCompare.wdl:
```diff
@@ -0,0 +1,581 @@
+    }
+
+
+    File interval_list = select_first([interval_list_override, "gs://concordance/interval_lists/chr9.hg38.interval_list"])
```
can you just set these as defaults in the input section, instead of having this `select_first`?

kachulis commented on Fingerprints/FindSamplesToCompare.wdl:
```diff
@@ -0,0 +1,581 @@
+    Map[File, String] truthLabels    = lablesMap.map
+    Map[File, File]   truthIndex     = indexesMap.map
+
+    #call SwitchFilterAnnotation {
```
remove commented code?

kachulis commented on Fingerprints/FindSamplesToCompare.wdl:
```diff
@@ -0,0 +1,581 @@
+                docker = docker,
+                picard_jar = picard_cloud_jar
+        }
+        String notLabel="NOT_" + interval_and_label.right
```
this is cool!

kachulis commented on Fingerprints/FindSamplesToCompare.wdl:
```diff
@@ -0,0 +1,581 @@
+    File runs_file = select_first([runs_file_override, "gs://concordance/hg38/runs.conservative.bed"])
+    
+
+    Int VCF_disk_size = 50
```
maybe use `size(input_callset, "GB")` with some added overhead instead?

kachulis commented on BenchmarkVCFs/BenchmarkVCFs.wdl:
```diff
@@ -56,6 +58,8 @@ workflow Benchmark {
         passingOnly: {description:"Have vcfEval only consider the passing variants"}
         vcfScoreField: {description:"Have vcfEval use this field for making the roc-plot. If this is an info field (like VSQLOD) it should be provided as INFO.VQSLOD, otherewise it is assumed to be a format field."}
+        gatkJarForAnnotation: {description:"GATK jar that can calculate necessary annotations for jexl Selections when using VCFEval."}
+        annotationName: {description:"Annotation argument to GATK (-A argument)"}
```
cool!

kachulis commented on Fingerprints/FindSamplesToCompare.wdl:
```diff
@@ -0,0 +1,581 @@
+version 1.0
+import "../BenchmarkVCFs/BenchmarkVCFs.wdl" as Benchmark
+
+workflow FindSamplesToCompare {
```
up to you, but I feel like this name sells this workflow short!  it's doing a bit more than just finding samples to compare.

kachulis replied:
also should this go in the BenchmarkVCFs directory?  I feel like it really is a benchmarking workflow, which uses fingerprinting, as opposed to a fingerprinting workflow that also does some benchmarking on the side

kachulis reviewed (commented):
oh cool!  I was just thinking about the sample matching version of this.  And it looks like there's some new fun bells and whistles since I last saw it :)

meganshand commented:
@kachulis Thanks for the review! I'd really like to test this in Terra, but I'm not seeing this branch show up in dockstore. This is where I was expecting it to show up: https://dockstore.org/my-workflows/github.com/broadinstitute/palantir-workflows/FindSamplesToCompare 

Any thoughts on how to fix this?

kachulis commented:
hmm, that's where I would expect to see it... Is it under the "Unpublished" tab for you?  They show up in "Unpublished" first, so that's where I'd expect it to be, though I'm not seeing it on my account... but maybe you will?

kachulis commented on .dockstore.yml:
```diff
@@ -24,3 +24,6 @@ workflows:
     primaryDescriptorPath: /ImputationPipeline/Validation/FullImputationPRSValidation.wdl
+  - name: FindSamplesToCompare
+    subclass: WDL
+    primaryDescriptorPath: /Fingerprints/FindSamplesToCompare.wdl
```
ohhhh I think maybe this needs a new line at the end?

kachulis replied:
ha, or guess not!

meganshand commented:
Ahha, it's there now! I think a combination of a typo and my impatience were to blame. I'm going to test this in Terra before merging.

meganshand commented:
@kachulis I believe I've addressed your comments, could you please take another look? It ran successfully in Terra. Let me know if you have a better name. I went with FindSamplesAndBenchmark, but I'm very open to suggestions.

kachulis commented on BenchmarkVCFs/FindSamplesAndBenchmark.wdl:
```diff
@@ -0,0 +1,544 @@
+    }
+
+    Int VCF_disk_size = ceil(size(input_callset, "GiB")) + 10
+    File monitoring_script="gs://broad-dsde-methods-monitoring/cromwell_monitoring_script.sh"
```
this means that if you don't have access to this bucket you can't run the workflow, right?  I think better to make it an input to the workflow, so someone w/o access can still use.  And adding a comment that this script can be found in the `Scripts/monitoring` directory of this repo

meganshand replied:
Done

kachulis reviewed (changes requested):
looks good, just a few requests about making this more flexible for future use/others.  

And I like the new name much better!

meganshand commented on BenchmarkVCFs/FindSamplesAndBenchmark.wdl:
```diff
@@ -0,0 +1,544 @@
+
+        File? interval_list_override = "gs://concordance/interval_lists/chr9.hg38.interval_list"
+        File? runs_file_override = "gs://concordance/hg38/runs.conservative.bed"
+        String comparison_docker
```
Deleted this and the tasks themselves.

meganshand commented on BenchmarkVCFs/FindSamplesAndBenchmark.wdl:
```diff
@@ -0,0 +1,544 @@
+        Boolean remove_symbolic_alleles=false
+
+        Array[File] stratIntervals = [
+                         "gs://concordance/hg38/runs.conservative.bed",
```
Good call, I've removed them.

kachulis commented on BenchmarkVCFs/FindSamplesAndBenchmark.wdl:
```diff
@@ -0,0 +1,420 @@
+
+        Array[File] stratIntervals
+        Array[String] stratLabels = [
+                         "HMER_7_and_up",
```
since there's no default for the intervals, there shouldn't be a default for the interval labels

kachulis commented on BenchmarkVCFs/FindSamplesAndBenchmark.wdl:
```diff
@@ -0,0 +1,420 @@
+
+        String docker 
+
+        File? interval_list_override
```
this isn't used either, is it?

kachulis commented on BenchmarkVCFs/FindSamplesAndBenchmark.wdl:
```diff
@@ -0,0 +1,420 @@
+
+        Array[String] jexlVariantSelectors = ["vc.isSimpleIndel()  && vc.getIndelLengths().0<0", "vc.isSimpleIndel() && vc.getIndelLengths().0>0"]
+        Array[String] variantSelectorLabels = ["deletion","insertion"]
+        # Input for monitoring_script can be found here:https://github.com/broadinstitute/palantir-workflows/blob/main/Scripts/monitoring/cromwell_monitoring_script.sh.
```
space between "here:" and url

kachulis reviewed (approved):
@meganshand a couple more minor nitpicking requests, but otherwise looks good!  Merge away once these last ones are addressed.

Thanks for adding this!
